pageserver/src/tenant/timeline.rs

```diff
@@ -3242,7 +3242,11 @@ impl Timeline {
             // keys from `keyspace`, we expect there to be no overlap between it and the image covered key
             // space. If that's not the case, we had at least one key encounter a gap in the image layer
             // and stop the search as a result of that.
-            let removed = keyspace.remove_overlapping_with(&image_covered_keyspace);
+            let mut removed = keyspace.remove_overlapping_with(&image_covered_keyspace);
+            // Do not fire missing key error for sparse keys.
+            removed.remove_overlapping_with(&KeySpace {
+                ranges: vec![SPARSE_RANGE],
```

Comment:
Strictly speaking we don't have to remove the entire `SPARSE_RANGE` here, but only the inherited ones, right?

Reply:
Yes, but note that non-inherited one is already removed before, so at this point, keyspace only contains inherited ones.

Reply:
> so at this point, keyspace only contains inherited ones.

Yeah, that was my point. I don't think it makes a difference with respect to `remove_overlapping_with` performance though.